File: python/sync_identification_example.py
import base64
import requests


def encode_files(file_names):
    files_encoded = []
    for file_name in file_names:
        with open(file_name, "rb") as file:
            files_encoded.append(base64.b64encode(file.read()).decode("ascii"))
    return files_encoded


def identify_plant(file_names):
    images = encode_files(file_names)

    # see the docs for more optional attributes
    params = {
        "api_key": "-- ask for one: https://forms.gle/yK1AY53YkYJjsc8X8 --",
        "images": images,
        "modifiers": ["crops_fast", "similar_images"],
        "plant_language": "en",
        "plant_details": ["common_names",
                          "url",
                          "name_authority",
                          "wiki_description",
                          "taxonomy",
                          "synonyms"],
        }

    headers = {
        "Content-Type": "application/json"
        }

    response = requests.post("https://api.plant.id/v2/identify",
                             json=params,
                             headers=headers)

    return response.json()


if __name__ == '__main__':
    print(identify_plant(["photo1.jpg", "photo2.jpg"]))

File: ruby/sync_identification_example.rb
require 'uri'
require 'net/http'
require 'json'

api_key = "YOUR_KEY_HERE"
base64_image = "YOUR_BASE64_STRING_HERE"
images = [base64_image]
modifiers = ["crops_fast", "similar_images"]
plant_language = "en"
plant_details = ["common_names", "url", "name_authority", "wiki_description", "taxonomy", "synonyms"]

uri = URI('https://api.plant.id/v2/identify')
https = Net::HTTP.new(uri.host, uri.port)
https.use_ssl = true

request = Net::HTTP::Post.new(uri.path)

data = {
  "images": images,
  "modifiers": modifiers,
  "plant_details": plant_details,
  "api_key": api_key
}.to_json
request.body = data

# API key can be pass either in the body or as a headers
# request["Api-Key"] = api_key
request["Content-Type"] = "application/json"
response = https.request(request)
json_raw =  response.body
p JSON.parse(json_raw)
